src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,494 @@
+    pipelineConfigurationBody: openSearchTemplate,
+    pipelineName: "dynamodb-integration-2",
+    logPublishingOptions: {
+      isLoggingEnabled: true,
+      cloudWatchLogDestination: {
+        logGroup: logGroup.logGroupName,
+      },
+    },
+  }
+);
+
+```
+
+After deploying the resources, you can test the data ingestion process by adding an item to the `Todo` table. However, before doing that, let's verify that the pipeline has been set up correctly.
+
+In the AWS console, navigate to OpenSearch and then to the pipelines section. You should find your configured pipeline and review its settings to ensure they match your expectations:
+
+![A screenshot displaying the OpenSearch OSIS pipeline created under the DynamoDB integrations section](/images/gen2/opensearch-integration/OpenSearch_pipeline.png)
+
+You can also check this in the DynamoDB console by going to the Integrations section of the tables.
+
+![A screenshot displaying the OpenSearch OSIS pipeline created within the 'Ingestion -> Pipelines' section of the OpenSearch Console.](/images/gen2/opensearch-integration/OpenSearch_DynamoDB_integration.png)
+
+## Step 4: Expose new queries on OpenSearch
+
+### Step 4a:Add OpenSearch Datasource to backend
+
+First, Add the OpenSearch data source to the data backend.
+
+``` ts title="amplify/backend.ts"
+// Add OpenSearch data source 
+const osDataSource = backend.data.addOpenSearchDataSource(
+  "osDataSource",
+  openSearchDomain
+);
+```
+### Step 4b: Create Resolver and attach to query
+
+Let's create the search resolver. Create a new file named `amplify/data/searchBlogResolver.ts` and paste the following code. For additional details please refer to [Amazon OpenSearch Service Resolvers](https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-elasticsearch-resolvers-js.html)
+
+```ts title="amplify/data/searchBlogResolver.ts"
+
+import { util } from "@aws-appsync/utils";
+
+/**
+ * Searches for documents by using an input term
+ * @param {import('@aws-appsync/utils').Context} ctx the context
+ * @returns {*} the request
+ */
+
+export function request(ctx) {
+  return {
+    operation: "GET",
+    path: "/todo/_search",
+  };
+}
+
+/**
+ * Returns the fetched items
+ * @param {import('@aws-appsync/utils').Context} ctx the context
+ * @returns {*} the result
+ */
+
+export function response(ctx) {
+  if (ctx.error) {
+    util.error(ctx.error.message, ctx.error.type);
+  }
+  return ctx.result.hits.hits.map((hit) => hit._source);
+}
+
+
+```
+
+### Step 4c: Add the AppSync Resolver for the Search Query
+
+Update the schema and add a searchTodo query.
+
+``` ts title="amplify/data/resource.ts"
+
+const schema = a.schema({
+  Todo: a
+    .model({
+      content: a.string(),
+      done: a.boolean(),
+      priority: a.enum(["low", "medium", "high"]),
+    })
+    .authorization([a.allow.public()]),
+
+  //highlight-start  
+    searchTodos: a
+    .query()
+    .returns(a.ref("Todo").array())
+    .authorization([a.allow.public()])
```

Comment:
do sweep to make sure the new authZ format is used: "allow => allow.publicApiKey()"

Reply:
corrected the auth format.